cmd/agent/app/reporter/grpc/builder_test.go

```diff
@@ -149,7 +149,7 @@ func TestBuilderWithCollectors(t *testing.T) {
 			cfg.Notifier = test.notifier
 			cfg.Discoverer = test.discoverer
 
-			conn, err := cfg.CreateConnection(zap.NewNop())
+			conn, err := cfg.CreateConnection(zap.NewNop(), metrics.NullFactory)
```

Comment:
Please add a test for the new metrics. May need to refactor the builder a bit for better testability.